aries_cloudagent/protocols/revocation_notification/v1_0/messages/revoke.py

```diff
@@ -0,0 +1,48 @@
+"""Revoke message."""
+
+from marshmallow import fields
+from .....messaging.agent_message import AgentMessage, AgentMessageSchema
+from .....messaging.valid import UUIDFour
+from ..message_types import PROTOCOL_PACKAGE, REVOKE
+
+HANDLER_CLASS = f"{PROTOCOL_PACKAGE}.handlers.revoke_handler.RevokeHandler"
+
+
+class Revoke(AgentMessage):
+    """Class representing revoke message."""
+
+    class Meta:
+        """Revoke Meta."""
+
+        handler_class = HANDLER_CLASS
+        message_type = REVOKE
+        schema_class = "RevokeSchema"
+
+    def __init__(self, *, thread_id: str, comment: str = None, **kwargs):
+        """Initialize revoke message."""
+        super().__init__(**kwargs)
+        # TODO support please ack
+        self.thread_id = thread_id
+        self.comment = comment
+
+
+class RevokeSchema(AgentMessageSchema):
+    """Schema of Revoke message."""
+
+    class Meta:
+        """RevokeSchema Meta."""
+
+        model_class = Revoke
+
+    # TODO support please ack
+    thread_id = fields.Str(
+        required=True,
+        description=(
+            "Thread ID of credential exchange resulting in this issued credential"
+        ),
+        example=UUIDFour.EXAMPLE,
+    )
```

Comment:
I suggest we add the revocation registry id and rev index to this notification.  The holder *may* not know the thread_id (if they don't store it when they receive the credential) but they will know the rev reg id and index because it will be part of the credential.

Reply:
Similar feelings here. I agree that these values would be much more useful than the thread ID but I hesitate to deliberately diverge from the spec without calling out that we are at least.

Reply:
@swcurran what are your thoughts on this?

Reply:
> what are your thoughts on this?

I add them to the thread above, when I meant add them to this one :-). See above ^^^.